src/SchemaRegistry.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.21;
+
+import { Schema } from "./struct/Schema.sol";
+
+contract SchemaRegistry {
+  mapping(bytes32 id => Schema schema) public schemas;
+  uint256 public numberOfSchemas = 0;
+
+  error SchemaAlreadyExists();
+  error SchemaNameMissing();
+  error SchemaStringMissing();
+
+  event SchemaCreated(string name, bytes32 schemaId);
+
+  function getIdFromSchemaString(string calldata _schema) public pure returns (bytes32) {
+    return keccak256(abi.encodePacked(_schema));
+  }
+
+  function createSchema(
+    string calldata name,
+    string calldata description,
+    string calldata context,
+    string calldata schemaString
+  ) public {
+    if (bytes(name).length == 0) {
+      revert SchemaNameMissing();
+    }
+
+    if (bytes(schemaString).length == 0) {
+      revert SchemaStringMissing();
+    }
+
+    bytes32 schemaId = getIdFromSchemaString(schemaString);
+
+    if (bytes(schemas[schemaId].name).length > 0) {
+      revert SchemaAlreadyExists();
+    }
+
+    schemas[schemaId] = Schema(name, description, context, schemaString);
+    numberOfSchemas++;
+    emit SchemaCreated(name, schemaId);
```

Comment:
better to have all schema metadata

Reply:
Just to double check, this means everything, except `schemaString`, right?
==> `event SchemaCreated(bytes32 id, string name, string description, string context)`